python/pyarrow/_dataset.pyx

```diff
@@ -773,6 +789,14 @@ cdef class FileFragment(Fragment):
         Fragment.init(self, sp)
         self.file_fragment = <CFileFragment*> sp.get()
 
+    def __reduce__(self):
+        buffer = self.buffer
+        return self.format.make_fragment, (
```

Comment:
By specifying here the method on a format object (`format.make_fragment`), it also automatically pickles the `format` *instance* ?

Reply:
We should maybe ensure this with testing a picking roundtrip for a case that specified read params in the ParquetFileFormat object

Reply:
Adding a case.

Reply:
> By specifying here the method on a format object (`format.make_fragment`), it also automatically pickles the `format` _instance_ ?

Apparently.